app/scripts/modules/core/src/api/mock/mockHttpUtils.ts

```diff
@@ -0,0 +1,40 @@
+import { $httpBackend } from 'ngimport';
+
+export type Verb = 'GET' | 'POST' | 'PUT' | 'PATCH' | 'DELETE';
+export type UrlArg = string | RegExp | ((url: string) => boolean);
+
+export interface IDeferred {
+  promise: Promise<any>;
+  resolve: (result: any) => void;
+  reject: (error: any) => void;
+  settled: boolean;
+}
+
+export function deferred() {
+  const deferredObj = {} as IDeferred;
+  deferredObj.promise = new Promise((resolve, reject) => {
+    deferredObj.settled = false;
+
+    deferredObj.resolve = (result) => {
+      deferredObj.settled = true;
+      resolve(result);
+    };
+
+    deferredObj.reject = (error) => {
+      deferredObj.settled = true;
+      reject(error);
+    };
+  });
+
+  return deferredObj;
+}
+
+export function kickAngularJS() {
```

Comment:
Kind of a nit, but I think calling this `flushAngularJS` or `flushHttpBackend` is more cohesive with the naming in the MockHttpClient, where it is also called flushing.

Reply:
you don't like my dig on angularjs? 😆